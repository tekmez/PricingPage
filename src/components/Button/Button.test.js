import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import CustomButton from './Button'
import styles from './Button.style'

describe('Touchable Button', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<CustomButton title="Click me" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const testTitle = 'Click me'
    const comp = render(<CustomButton title={testTitle} />)
    const buttonText = comp.getByTestId('button-title').children[0]
    expect(buttonText).toBe(testTitle)
  })

  it('should call onPress event', () => {
    const handlePress = jest.fn()
    const comp = render(<CustomButton onPress={handlePress} />)
    fireEvent.press(comp.getByTestId('button-touchable'))
    expect(handlePress).toBeCalled()
  })

  it('should render given theme style', () => {
    const testTheme = 'secondary'
    const comp = render(<CustomButton theme={testTheme} />)
    const buttonStyle = comp.getByTestId('button-touchable').props.style
    expect(buttonStyle).toMatchObject(styles[testTheme].button)
  })
})
