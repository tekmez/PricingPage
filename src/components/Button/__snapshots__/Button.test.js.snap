// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable Button should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#40A2E3",
      "borderRadius": 8,
      "opacity": 1,
      "paddingHorizontal": 50,
      "paddingVertical": 12,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "Inter-Regular",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
    testID="button-title"
  >
    Click me
  </Text>
</View>
`;
